subsys/bluetooth/host/smp.c

```diff
@@ -2111,6 +2138,26 @@ static u8_t send_pairing_rsp(struct bt_smp *smp)
 }
 #endif /* CONFIG_BT_PERIPHERAL */
 
+static u8_t smp_pairing_accept_query(struct bt_conn *conn,
+				    struct bt_smp_pairing *pairing)
+{
+#if defined(CONFIG_BT_SMP_APP_PAIRING_ACCEPT)
+	if (bt_auth && bt_auth->pairing_accept) {
+		struct bt_conn_pairing_feat feat = {
```

Comment:
I'd make this `const struct ...`  here to make it clear that this is not to be modified by the callback, and also make the pointer to the callback use `const`

Reply:
Can do that. Thought of it, but it doesn't really matter as long as it is isolated from the packet data.

Reply:
It might give the compiler more opportunities for optimisation though, right? At least I'd have thought for this reason it to be good practice to declare everything const that can be declared const.

Reply:
Anyway, I don't really have a super strong opinion on this, however as the callback is a public API we should avoid changing the types after it's merged upstream.

Reply:
Shouldn't we change `struct bt_conn *conn` to `const struct bt_conn *const conn` as well? Then we use the same practice.

Reply:
`bt_conn` is a little bit of a different story. It's an opaque type, i.e. the application has no visibility into its implementation and cannot do direct access to its members. It's also valid for the application to want to call some API which takes a bt_conn pointer from within the callback, in which case things would break (create a compiler error) since no bt_conn-taking APIs use const.

Reply:
Alright, just added const.